### PR TITLE
Add ignore for other IDEs and add ignore for docs/.venv/

### PR DESCRIPTION
**Pull request by Kureii**
I edit git ignore for other IDEs. This is the first step to add support for cmake build and make the project available to more developers.

### CODE REVIEW

**Comment by Kureii** [comment]
Are you okay with my plan?

File: .gitignore [hunk 1]
```diff
@@ -7,8 +7,30 @@ default/
...
 
-# IDE files
+# QT Creator
+*.user
+*.creator.*
+*.autosave
+*.pro.user
+*.qbs.user
 *.qbs.user*
+*.pro.user.*
```
Comment (bjorn) [inline .gitignore:17]: I don't see why we would need to ignore pro files again. You're not going to add back a qmake project, are you?

File: .gitignore [hunk 2]
```diff
@@ -7,8 +7,30 @@ default/
 # Mac stuff
 *.DS_Store
 
-# IDE files
+# QT Creator
+*.user
+*.creator.*
+*.autosave
+*.pro.user
+*.qbs.user
```
Comment (bjorn) [inline .gitignore:15]: This should be covered by the existing `*.qbs.user*` ignore rule, right?

File: .gitignore [hunk 3]
```diff
@@ -7,8 +7,30 @@ default/
 # Mac stuff
 *.DS_Store
 
-# IDE files
+# QT Creator
+*.user
```
Comment (bjorn) [inline .gitignore:11]: If we're going to ignore `*.user`, then a bunch of ignore rules below are not needed.

File: .gitignore [hunk 4]
```diff
@@ -7,8 +7,30 @@ default/
 # Mac stuff
 *.DS_Store
 
-# IDE files
+# QT Creator
```
Comment (bjorn) [inline .gitignore:10]:
```suggestion
# Qt Creator
```

**Review by bjorn — COMMENTED** [review]
Well, I'm curious what the CMake project would look like.

Note that Qbs can also generate Visual Studio project files (I haven't tested it) and that there is a Qbs extension for VS Code. So I think Qbs is quite available (though not sure about CLion or Xcode).

**Comment by bjorn** [comment]
Oh, and the commits from #4073 came back. Please rebase instead of merging upstream. :-)

**Comment by bjorn** [comment]
Hmm, you do need to resolve the conflicts before doing `git rebase --continue`. Alternatively, if you rebase using `git rebase -i upstream/master`, then an editor will open where you can drop the changes that were already applied upstream, so you don't need to resolve conflicts in them.

**Pull request closed without merging**